src/basic/signal-util.c

```diff
@@ -241,11 +240,14 @@ int signal_from_string(const char *s) {
                 s += 6;
                 offset = SIGRTMIN;
         }
-        if (safe_atou(s, &u) >= 0) {
-                signo = (int) u + offset;
-                if (SIGNAL_VALID(signo))
-                        return signo;
+
+        if (safe_atoi(s, &signo) >= 0) {
+                if (signo > 0 && signo < _NSIG - offset)
+                        return signo + offset;
+                else
+                        return -ERANGE;
         }
```

Comment:
hmm, while this patch will fix the issue at hand, we should probably clean this up more comprehensively and correctly... Specifically:

1. The valid for SIGRTMIN-based signals is SIGRTMIN…SIGRTMAX, and not SIGRTMIN…_NSIG...
2. We should accept "SIGRTMAX-x" too here, in addition to "SIGRTMIN+x"
3. We should accept "SIGRTMIN" as alias for "SIGRTMIN+0", and similar "SIGRTMAX" as alias for "SIGRTMAX+0"